### PR TITLE
[Demangling] Fix duplicate _gCRAnnotations symbol problems.

### PR DESCRIPTION
**Pull request by al45tair**
Moved the `_gCRAnnotations` declarations to their own object module, which will help to avoid duplicate symbol problems (at least with `.a` files).

Also tweaked things to make it so that the demangler and runtime versions of the message setting code will interoperate (and so that they'll interoperate better with other implementations that might creep in from somewhere, like the one in LLVMSupport).

rdar://91095592

### CODE REVIEW

**Comment by al45tair** [comment]
@swift-ci Please smoke test

File: stdlib/public/runtime/Errors.cpp [hunk 1]
```diff
@@ -254,47 +259,34 @@ void swift::printCurrentBacktrace(unsigned framesToSkip) {
...
-  }
-  
-  CRSetCrashLogMessage(newMessage);
+  char *oldMessage = nullptr;
+  char *newMessage = nullptr;
 
-  crashlogLock.unlock();
+  oldMessage = std::atomic_load_explicit(
+    (volatile std::atomic<char *> *)&gCRAnnotations.message,
+    std::memory_order_relaxed);
```
Comment (mikeash) [inline stdlib/public/runtime/Errors.cpp:272]: Not that it's going to matter much in practice, but this should be `SWIFT_MEMORY_ORDER_CONSUME`.

**Review by mikeash — CHANGES_REQUESTED** [review]

**Comment by al45tair** [comment]
@swift-ci Please smoke test

**Review by mikeash — APPROVED** [review]

**Comment by al45tair** [comment]
@swift-ci Please smoke test

**Comment by al45tair** [comment]
@swift-ci Please smoke test

**Comment by al45tair** [comment]
@swift-ci Please smoke test platform linux

**Comment by al45tair** [comment]
@swift-ci Please smoke test Linux platform

**Pull request merged**